src/harness/reference_models/dpa/dpa_mgr.py

```diff
@@ -495,26 +525,27 @@ def _DefaultProtectionPoints(name,
   This creates a default set of points by regular sampling of the contour and gridding
   the interior of the DPA zone. The contour and zone are separated in front and back,
   using the US border (with a buffer) as the delimitation.
+  If the DPA is a Point, a single point is returned however.
 
   Args:
-    num_pts_front_border: number of points in the front border
-    num_pts_back_border: number of points in the back border
-    num_pts_front_zone: number of points in the front zone
-    num_pts_back_zone: number of points in the back zone
-    front_us_border_buffer_km: buffering of US border for delimiting front/back.
+    dpa_geometry: The DPA geometry as a |shapely.Polygon or Point|.
+    num_pts_front_border: Number of points in the front border.
+    num_pts_back_border: Number of points in the back border.
+    num_pts_front_zone: Number of points in the front zone.
+    num_pts_back_zone: Number of points in the back zone.
+    front_us_border_buffer_km: Buffering of US border for delimiting front/back.
   """
-  us_border = zones.GetUsBorder()
-  dpa_zone = zones.GetDpaZones()[name]
   # Case of DPA points
-  if isinstance(dpa_zone, sgeo.Point):
-    return [ProtectionPoint(longitude=dpa_zone.x,
-                            latitude=dpa_zone.y)]
+  if isinstance(dpa_geometry, sgeo.Point):
+    return [ProtectionPoint(longitude=dpa_geometry.x, latitude=dpa_geometry.y)]
 
+  # Case of Polygon
+  us_border = zones.GetUsBorder()
   us_border_ext = us_border.buffer(front_us_border_buffer_km / 111.)
```

Comment:
"front_us_border_buffer_km / 111." will convert the buffer to degrees?

Reply:
Yes, 111 would be the equatorial distance for one degree.
Have a look at the new code, where I use the US centroid of 45 degrees instead.